src/_pytest/stepwise.py

```diff
@@ -119,4 +121,6 @@ def pytest_report_collectionfinish(self) -> Optional[str]:
         return None
 
     def pytest_sessionfinish(self) -> None:
+        if hasattr(self.config, "workerinput"):
```

Comment:
```suggestion
        # Do not update cache if this process is a xdist worker to prevent race conditions (#10641).
        if hasattr(self.config, "workerinput"):
```

Reply:
Your feedback has been incorporated into the code